internal/controller/organization_controller_test.go

```diff
@@ -116,15 +114,29 @@ var _ = Describe("Organization controller", func() {
 			By("Deleting the first organization")
 			Expect(k8sClient.Delete(ctx, org1)).Should(Succeed())
 
-			_, err = reconciler.Reconcile(ctx, reconcile.Request{
-				NamespacedName: types.NamespacedName{Name: "test-1"},
-			})
-			Expect(err).NotTo(HaveOccurred())
+			Eventually(func() error {
+				err := k8sClient.Get(ctx, client.ObjectKey{Name: "test-1"}, &securityv1alpha1.Organization{})
+				if errors.IsNotFound(err) {
+					return nil
+				}
+				if err != nil {
+					return err
+				}
```

Comment:
I think you can use this instead.

```suggestion
				Expect(err).NotTo(HaveOccurred())
```